core/server/api_container/server/startosis_engine/kurtosis_instruction/kurtosis_instruction.go

```diff
@@ -34,5 +35,11 @@ type KurtosisInstruction interface {
 	ValidateAndUpdateEnvironment(environment *startosis_validator.ValidatorEnvironment) error
 
 	// TryResolveWith assesses whether the instruction can be resolved with the one passed as an argument.
-	TryResolveWith(other KurtosisInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+	TryResolveWith(other EnclavePlanInstruction, enclaveComponents *enclave_structure.EnclaveComponents) enclave_structure.InstructionResolutionStatus
+
+	GetCapabilites() EnclavePlanCapabilitiesProvider
```

Comment:
Can we be more explicit here? given that there's already `*Capablities` object for each instruction, this broad name is verrrry confusing